Added tests for functions in node_printers.py

shirleyyzhang opened the pull request:
## Proposed Changes

_(Describe your changes here. Also describe the motivation for your changes: what problem do they solve, or how do they improve the application or codebase? If this pull request fixes an open issue, [use a keyword to link this pull request to the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).)_

Added tests to increase coverage for functions in node_printers.py used by PycodestyleChecker. 
The tests added cover the functions that render the following PEP8 errors: E101, E116, E124, E128, E201, E202

<details>
<summary>Screenshots of your changes (if applicable)</summary>

</details>

## Type of Change

_(Write an `X` or a brief description next to the type or types that best describe your changes.)_

| Type                                                                                    | Applies? |
| --------------------------------------------------------------------------------------- | -------- |
| 🚨 _Breaking change_ (fix or feature that would cause existing functionality to change) |          |
| ✨ _New feature_ (non-breaking change that adds functionality)                          |          |
| 🐛 _Bug fix_ (non-breaking change that fixes an issue)                                  |          |
| ♻️ _Refactoring_ (internal change to codebase, without changing functionality)          |          |
| 🚦 _Test update_ (change that _only_ adds or modifies tests)                            |     X     |
| 📚 _Documentation update_ (change that _only_ updates documentation)                    |          |
| 📦 _Dependency update_ (change that updates a dependency)                               |          |
| 🔧 _Internal_ (change that _only_ affects developers or continuous integration)         |          |

## Checklist

_(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)_

Before opening your pull request:

- [x] I have performed a self-review of my changes.
  - Check that all changed files included in this pull request are intentional changes.
  - Check that all changes are relevant to the purpose of this pull request, as described above.
- [x] I have added tests for my changes, if applicable.
  - This is **required** for all bug fixes and new features.
- [x] I have updated the project documentation, if applicable.
  - This is **required** for new features.
- [x] I have updated the project Changelog (this is required for all changes).
- [x] If this is my first contribution, I have added myself to the [list of contributors](https://github.com/pyta-uoft/pyta/blob/master/README.md#contributors).

After opening your pull request:

- [x] I have verified that the pre-commit.ci checks have passed.
- [x] I have verified that the CI tests have passed.
- [x] I have reviewed the test coverage changes reported by Coveralls.
- [x] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments

_(Include any questions or comments you have regarding your changes.)_

For the test covering the PEP8 error E101, which triggers when "indentation contains mixed spaces and tabs", pycodestyle will also always emit PEP8 error W191 (“indentation contains tabs”). As a result, any case that triggers E101 will also emit W191, so the test will consistently raise an AssertionError.

Currently, indentation in line 2 of e101_error.py consists of spaces, and indentation in line 3 consists of a tab character:
```python
if True:
    # Indentation contains a space followed by a tab
	pass
```
which triggers E101 (expected) and W191, causing the AssertionError. 

I have considered:
-  suppressing W191 for the test, but it is not a sustainable fix. 
-  adding W191 to the expected messages for the test. However, the expected id is always taken from msg_id in error_params, so I will need to investigate further. 
- modifying the checker to expect (and/or ignore) W191 when E101 is expected. 
Any suggestion would be appreciated!

david-yz-liu commented on CHANGELOG.md:
```diff
@@ -15,6 +15,12 @@ and adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### 🔧 Internal changes
 
+## [2.11.2] - 2025-09-12
```
You should not be adding a new release section; instead, add this entry just under the unreleased section

david-yz-liu commented on README.md:
```diff
@@ -120,4 +120,5 @@ Lana Wehbeh,
 Jasmine Wu,
 Raine Yang,
 Philippe Yu,
+Shirley Zhang
```
Add a comma after your name

david-yz-liu reviewed (commented):
@shirleyyzhang nice work! To address your question, I think the suggestion from yesterday's meeting to just create a separate test case for E101 is a good one; so rather than including this one in the `error_params` variable, you can create a test case with the same structure as the existing one, but that has specific functionality for the E101 check. Make sure to include in the test case docstring the reason for why a separate test case is used.

david-yz-liu commented on tests/test_custom_checkers/test_pycodestyle_checker.py:
```diff
@@ -39,6 +45,7 @@
     ("E301", [(5, 4, "expected 1 blank line, found 0")]),
     ("E303", [(6, 0, "too many blank lines (3)")]),
     ("E304", [(12, 0, "blank lines found after function decorator")]),
+    ("E306", [(3, 4, "expected 1 blank line before a nested definition, found 0")]),
```
@shirleyyzhang sorry I missed this in my initial review, but I don't think you intended to add this line (as there are no example files for this error)

shirleyyzhang commented:
> @shirleyyzhang nice work! To address your question, I think the suggestion from yesterday's meeting to just create a separate test case for E101 is a good one; so rather than including this one in the `error_params` variable, you can create a test case with the same structure as the existing one, but that has specific functionality for the E101 check. Make sure to include in the test case docstring the reason for why a separate test case is used.

Hi @david-yz-liu, thanks for the clarification!

Here are the changes I made:
- Fixed CHANGELOG and README modifications
- Added 2 test cases in test_pystylecode_checker.py specifically for E101:

```python
def test_e101_error(self):
        """
        Test case for PEP8 error E101 indentation contains mixed spaces and tabs

        This test is separate from test_error_cases() to handle the W191 message that is
        always emitted alongside E101.
        """
        expected = [
            ("E101", 3, 0, "indentation contains mixed spaces and tabs"),
            ("W191", 3, 0, "indentation contains tabs"),
        ]
        mod = MANAGER.ast_from_file(os.path.join(DIR_PATH, "e101_error.py"))
        with self.assertAddsMessages(
            *[
                pylint.testutils.MessageTest(
                    msg_id="pep8-errors",
                    line=line,
                    args=(msg_id, f"line {line}, column {col}: {desc}"),
                )
                for msg_id, line, col, desc in expected
            ],
            ignore_position=True,
        ):
            self.checker.process_module(mod)

def test_e101_no_error(self):
        """
        Tests that PEP8 error E101 is not triggered.
        This test is separate due to the same issue as noted in test_e101_error().
        """
        mod = MANAGER.ast_from_file(os.path.join(DIR_PATH, "e101_no_error.py"))
        with self.assertNoMessages():
            self.checker.process_module(mod)
```

These tests:
- solve the aforementioned issue with W191
- have similar structure to the existing test
I chose not to use pytest's parametrize decorator because it resulted in some excessive args unpacking (since the test only runs for one file and asserts that both messages are emitted together in that run anyways). I also left ```expected``` as a list of tuples for the same reason. 

Let me know if I should make further adjustments!

david-yz-liu reviewed (approved):
Nice work, @shirleyyzhang!
